llama_index/llama_dataset/generator.py

```diff
@@ -87,30 +91,34 @@ def __init__(
     def from_documents(
         cls,
         documents: List[Document],
-        service_context: ServiceContext | None = None,
+        llm: Optional[LLM] = None,
+        transformations: Optional[List[TransformComponent]] = None,
         num_questions_per_chunk: int = 3,
-        text_question_template: BasePromptTemplate | None = None,
-        text_qa_template: BasePromptTemplate | None = None,
-        question_gen_query: str | None = None,
-        required_keywords: List[str] | None = None,
-        exclude_keywords: List[str] | None = None,
+        text_question_template: Optional[BasePromptTemplate] = None,
+        text_qa_template: Optional[BasePromptTemplate] = None,
+        question_gen_query: Optional[str] = None,
+        required_keywords: Optional[List[str]] = None,
+        exclude_keywords: Optional[List[str]] = None,
         show_progress: bool = False,
         workers: int = DEFAULT_NUM_WORKERS,
+        # deprecated
+        service_context: Optional[ServiceContext] = None,
     ) -> RagDatasetGenerator:
         """Generate dataset from documents."""
-        if service_context is None:
-            service_context = service_context or ServiceContext.from_defaults(
-                chunk_size_limit=3000
```

Comment:
this is technically a behavior change right

Reply:
Kind of? Let me see if I can pass a prompt helper in